src/viam/app/data_client.py

```diff
@@ -298,6 +300,39 @@ async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes
         response: TabularDataByMQLResponse = await self._data_client.TabularDataByMQL(request, metadata=self._metadata)
         return [bson.decode(bson_bytes) for bson_bytes in response.raw_data]
 
+    async def get_latest_tabular_data(self, part_id: str, resource_name: str, resource_subtype: str, method_name: str) -> Optional[Tuple[Dict[str, ValueTypes], datetime, datetime]]:
+        """Gets the most recent tabular data captured from the specified data source, as long as it was synced within the last year.
+
+        ::
+
+                time_captured, time_synced, payload = await data_client.get_latest_tabular_data(
+                    part_id="<PART-ID>",
+                    resource_name="<RESOURCE-NAME>",
+                    resource_subtype="<RESOURCE-SUBTYPE>",
+                    method_name="<METHOD-NAME>"
+                )
+
+
+        Args:
+            part_id (str): The ID of the part that owns the data.
+            resource_name (str): The name of the requested resource that captured the data.
+            resource_subtype (str): The subtype of the requested resource that captured the data.
+            method_name (str): The data capture method name.
+
+        Returns:
+            Optional[Tuple[Dict[str, ValueTypes], datetime, datetime]]: A tuple which is None data hasn't been synced yet for the data source, otherwise the tuple contains the following:
```

Comment:
"A return value of None means that data hasn't been synced yet for the data source or the most recently captured data was over a year ago"